Controls/Mail/Recipient.c

```diff
@@ -10,14 +11,33 @@
  */
 void readMail(void) {
 	OS_ERR err; // Make sure to check for errors and print the error code if not OS_ERR_NONE
+	printf("%s\n",mailbox);
+	OS_OPT opt =OS_OPT_TIME_HMSM_STRICT;
+	OSTimeDlyHMSM(0,0,2,0, opt,&err);//TWO seconds
+		if(err!=OS_ERR_NONE){
+			printf("Error from OSTimeDlyHMSM Recipient, readMail");
+			return;
+		}
 }
 
 /**
  * @brief Waits on the mailbox to be signaled. Once the mailbox has signaled, call readMail. Loop
  */
 void Task_Recipient(void* p_arg) {
 	OS_ERR err;	// Make sure to check for errors and print the error code if not OS_ERR_NONE
+	CPU_TS timestamp;
 	while(1){
-		
+		OSSemPend(
+			&MailboxFlag_Sem4,
+			0,
+			OS_OPT_PEND_BLOCKING,
+			&timestamp,
+			&err
+			);//pending on mail--is ther any mail yet??
```

Comment:
Does this need to be on multiple lines? Make it consistent with the way you call similar functions throughout the other files.